src/deadline/job_attachments/_diff.py

```diff
@@ -113,3 +114,50 @@ def compare_manifest(
             differences.append((FileStatus.DELETED, manifest_path))
 
     return differences
+
+
+def _fast_file_list_to_manifest_diff(
+    root: str, current_files: List[str], diff_manifest: BaseAssetManifest, logger: ClickLogger
+) -> List[str]:
+    """
+    Perform a fast difference of the current list of files to a previous manifest to diff against using time stamps and file sizes.
+    :param root: Root folder of files to diff against.
+    :param current_files: List of files to compare with.
+    :param diff_manifest: Manifest containing files to diff against.
+    :return List[str]: List of files that are new, or modified.
+    """
+    changed_paths: List[str] = []
+    input_files_map: Dict[str, BaseManifestPath] = {}
+    for input_file in diff_manifest.paths:
+        # Normalize paths so we can compare different OSes
+        normalized_path = os.path.normpath(input_file.path)
+        input_files_map[normalized_path] = input_file
+
+    # Iterate for each file that we found in glob.
+    for local_file in current_files:
+        # Get the file's time stamp and size. We want to compare both.
+        # From enabling CRT, sometimes timestamp update can fail.
+        local_file_path = Path(local_file)
+        file_stat = local_file_path.stat()
+
+        # Compare the glob against the relative path we store in the manifest.
+        root_relative_path = str(PurePosixPath(*local_file_path.relative_to(root).parts))
+        if root_relative_path not in input_files_map:
+            # This is a new file
+            logger.echo(f"Found difference at: {root_relative_path}, Status: FileStatus.NEW")
+            changed_paths.append(local_file)
+        else:
+            # This is a modified file, compare with manifest relative timestamp.
+            input_file = input_files_map[root_relative_path]
+            # Check file size first as it is easier to test. Usually modified files will also have size diff.
+            if file_stat.st_size != input_file.size:
+                changed_paths.append(local_file)
+                logger.echo(
+                    f"Found size difference at: {root_relative_path}, Status: FileStatus.MODIFIED"
+                )
+            elif int(file_stat.st_mtime_ns // 1000) != input_file.mtime:
+                changed_paths.append(local_file)
+                logger.echo(
+                    f"Found time difference at: {root_relative_path}, Status: FileStatus.MODIFIED"
```

Comment:
nit - If we want the FileStatus visibility, could we pass by reference rather than hard code them? If any of that status gets changed, this can easily get out of sync.